hw/hal/src/hal_common.c

```diff
@@ -57,3 +57,27 @@ hal_spi_data_mode_breakout(uint8_t data_mode, int *out_cpol, int *out_cpha)
         return -1;
     }
 }
+
+const char *
+hal_reset_cause_str(void)
+{
+    enum hal_reset_reason cause;
+
+    cause = hal_reset_cause();
+    switch (cause) {
```

Comment:
could not this just be `switch (hal_reset_cause()) {` eliminating the var declaration?

Reply:
I often use local variables to make the code easier to read. It's up to compiler to optimize them out :)